synfig-core/src/modules/mod_magickpp/mptr_magickpp.cpp

```diff
@@ -128,33 +128,43 @@ magickpp_mptr::get_frame(synfig::Surface& surface, const synfig::RendDesc& /*ren
 		const auto height = image.size().height();
 		surface.set_wh(width, height);
 
-		image.type(Magick::TrueColorType);
-// Sadly Magick++ API is a mess with namespaces...
-{
-	using namespace Magick;
-	using namespace MagickCore;
+#if MagickLibVersion >= 0x701 // MAGICKCORE_CHECK_VERSION(7,0,0) does not work!
+		const bool has_alpha = image.alpha();
+#else
+		const bool has_alpha = image.matte();
+#endif
 
+		if (has_alpha)
 #if MagickLibVersion >= 0x701 // MAGICKCORE_CHECK_VERSION(7,0,0) does not work!
-		bool has_alpha = image.alpha();
+			image.type(Magick::TrueColorAlphaType);
 #else
-		bool has_alpha = image.matte();
+			image.type(Magick::TrueColorMatteType);
 #endif
+		else
+			image.type(Magick::TrueColorType);
+
 		const auto packet = image.getConstPixels(0, 0, width, height);
 
 		if (!packet) {
 			synfig::error(_("Magick++ importer: couldn't get pixel packet"));
 			return false;
 		}
+// Sadly Magick++ API is a mess with namespaces...
+{
+	using namespace Magick;
+	using namespace MagickCore;
+
 
 		constexpr synfig::Color::value_type factor = QuantumRange;
 
 		for (size_t y = 0, i = 0; y < height; ++y) {
 			for (size_t x = 0; x < width; ++x, ++i) {
```

Comment:
It looks like we can make this code clearer by taking `++i` out of the loop:
3 and 4 constants should be clearer than 2 and 3.
```
		for (size_t y = 0, i = 0; y < height; ++y) {
			for (size_t x = 0; x < width; ++x) {
#if MagickLibVersion >= 0x701 // MAGICKCORE_CHECK_VERSION(7,0,0) does not work!
				if (has_alpha) {
					surface[y][x] = synfig::Color(packet[i] / factor, packet[i+1] / factor, packet[i+2] / factor, packet[i+3] / factor);
					i += 4;
				} else {
					surface[y][x] = synfig::Color(packet[i] / factor, packet[i+1] / factor, packet[i+2] / factor, 1.);
					i += 3;
				}
#else
				const auto& color = packet[++i];
				if (has_alpha)
					surface[y][x] = synfig::Color(color.red / factor, color.green / factor, color.blue / factor, 1. - color.opacity / factor);
				else
					surface[y][x] = synfig::Color(color.red / factor, color.green / factor, color.blue / factor, 1.);
#endif
```

Reply:
Indeed, it's way better!
I just changed the pre-increment to post-increment on `else` branch, as it would skip `packet[0]` otherwise

Reply:
> as it would skip packet[0] otherwise

Good catch!